qiskit/algorithms/linear_solvers/numpy_linear_solver.py

```diff
@@ -0,0 +1,97 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""The Numpy LinearSolver algorithm (classical)."""
+
+from typing import List, Union, Optional, Callable
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.quantum_info import Operator, Statevector
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from .linear_solver import LinearSolverResult, LinearSolver
+from .observables.linear_system_observable import LinearSystemObservable
+
+
+class NumPyLinearSolver(LinearSolver):
+    """
+    The Numpy LinearSolver algorithm (classical).
+
+    This linear system solver computes the exact value of the given observable(s) or the full
+    solution vector if no observable is specified.
+
+    Examples:
+
+        .. jupyter-execute::
+
+            import numpy as np
+            from qiskit.algorithms.numpy_linear_solver import NumPyLinearSolver
+            from qiskit.algorithms.linear_solvers.matrices.tridiagonal_toeplitz import \
+             TridiagonalToeplitz
+            from qiskit.algorithms.linear_solvers.observables.matrix_functional import \
+             MatrixFunctional
+
+            matrix = TridiagonalToeplitz(2, 1, 1 / 3, trotter_steps=2)
+            right_hand_side = [1.0, -2.1, 3.2, -4.3]
+            observable = MatrixFunctional(1, 1 / 2)
+            rhs = right_hand_side / np.linalg.norm(right_hand_side)
+
+            np_solver = NumPyLinearSolver()
+            solution = np_solver.solve(matrix, rhs, observable)
+            result = solution.observable
```

Comment:
This is still failing CI, I guess jupyter sphinx isn't interpreting the `\` correctly

```suggestion
            import numpy as np
            from qiskit.algorithms.numpy_linear_solver import NumPyLinearSolver
            from qiskit.algorithms.linear_solvers.matrices import tridiagonal_toeplitz
            from qiskit.algorithms.linear_solvers.observables import matrix_functional

            matrix = tridiagonal_toeplitz.TridiagonalToeplitz(
                2, 1, 1 / 3, trotter_steps=2)
            right_hand_side = [1.0, -2.1, 3.2, -4.3]
            observable = matrix_functional.MatrixFunctional(1, 1 / 2)
            rhs = right_hand_side / np.linalg.norm(right_hand_side)

            np_solver = NumPyLinearSolver()
            solution = np_solver.solve(matrix, rhs, observable)
            result = solution.observable
```

If line length is a problem here lets just switch to module imports and use that.